dev-libs/boehm_gc/boehm_gc-7.6.4.recipe

```diff
@@ -0,0 +1,81 @@
+SUMMARY="The Boehm-Demers-Weiser conservative garbage collector"
+DESCRIPTION="The Boehm-Demers-Weiser conservative garbage collector can be \
+used as a garbage collecting replacement for C malloc or C++ new. It allows \
+you to allocate memory basically as you normally would, without explicitly \
+deallocating memory that is no longer useful. The collector automatically \
+recycles memory when it determines that it can no longer be otherwise \
+accessed."
+HOMEPAGE="http://www.hboehm.info/gc/"
+COPYRIGHT="1988, 1989 Hans-J. Boehm, Alan J. Demers
+	1991-1996 Xerox Corporation
+	1996-1999 Silicon Graphics
+	1999-2011 Hewlett-Packard Development Company, L.P."
+LICENSE="Boehm"
+REVISION="1"
+SOURCE_URI="https://github.com/ivmai/bdwgc/releases/download/v$portVersion/gc-$portVersion.tar.gz"
+CHECKSUM_SHA256="b94c1f2535f98354811ee644dccab6e84a0cf73e477ca03fb5a3758fb1fecd1c"
+SOURCE_DIR="gc-$portVersion"
+PATCHES="boehm_gc-$portVersion.patchset"
+
+ARCHITECTURES="?x86_gcc2 ?x86 x86_64"
+SECONDARY_ARCHITECTURES="?x86_gcc2 ?x86"
+
+portVersionCompat="$portVersion compat >= 7"
+
+PROVIDES="
+	boehm_gc$secondaryArchSuffix = $portVersionCompat
+	lib:libgc$secondaryArchSuffix = $portVersionCompat
+	lib:libcord$secondaryArchSuffix = $portVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libatomic_ops${secondaryArchSuffix}
+	"
+
+PROVIDES_devel="
+	boehm_gc${secondaryArchSuffix}_devel = $portVersionCompat
+	devel:libgc${secondaryArchSuffix} = $portVersionCompat
+	devel:libcord${secondaryArchSuffix} = $portVersionCompat
+	"
+REQUIRES_devel="
+	boehm_gc${secondaryArchSuffix} == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libatomic_ops$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	cmd:libtoolize$secondaryArchSuffix
+	cmd:autoconf
+	cmd:make
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:pkg_config$secondaryArchSuffix
+	"
+
+BUILD()
+{
+	libtoolize -fci
+	autoconf -f
+	runConfigure ./configure --enable-large-config \
+			--enable-threads=posix \
+			--enable-parallel-mark \
+			--disable-debug \
+			--disable-dependency-tracking
+	make $jobArgs
+}
+
+INSTALL()
+{
+	make install
+	prepareInstalledDevelLibs libgc libcord
+	fixPkgconfig
+	packageEntries devel $developDir
+}
+
+TEST()
+{
+	make check -e CFLAGS+="-fPIC"
```

Comment:
It builds fine on x86_64 and x86 (secondary arch), but when I run `haikuporter --test boehm_gc` (on x86_64) or `haikuporter --test boehm_gc_x86` (on x86_gcc2) I hit this error:
* on x86_64:
```
libtool: link: gcc -fexceptions -fPIC -o .libs/threadleaktest tests/thread_leak_test.o  ./.libs/libgc.so -L/packages/libatomic_ops-7.6.2-1/.self/develop/lib -latomic_ops -lpthread -Wl,-rpath -Wl,/packages/boehm_gc-7.6.4-1/.self/lib
tests/thread_leak_test.o: In function `main':
thread_leak_test.c:(.text+0xca): undefined reference to `GC_pthread_create'
thread_leak_test.c:(.text+0x128): undefined reference to `GC_pthread_join'
collect2: error: ld returned 1 exit status
```
* on x86 2ndary arch:
```
libtool: link: gcc -fexceptions -fPIC -o .libs/threadleaktest tests/thread_leak_test.o  ./.libs/libgc.so -L/packages/libatomic_ops_x86-7.6.2-1/.self/develop/lib/x86 -latomic_ops -lpthread -Wl,-rpath -Wl,/packages/boehm_gc_x86-7.6.4-1/.self/lib/x86
tests/thread_leak_test.o: In function `main':
thread_leak_test.c:(.text+0xd1): undefined reference to `GC_pthread_create'
thread_leak_test.c:(.text+0x129): undefined reference to `GC_pthread_join'
collect2: error: ld returned 1 exit status
```
I did not investigate yet.

Reply:
I believe having `CFLAGS+="-fPIC"` overriden the `CFLAGS` used during build. I've not tested but may be this would fix the issue:

```diff
-	make check -e CFLAGS+="-fPIC"
+	make check
```